op-guide/gc.md

```diff
@@ -0,0 +1,77 @@
+
+TiDB 采用 MVCC 的方式来进行并发控制。当对数据进行更新或者删除时，原有的数据不会被立刻删除，而是会被保留一段时间，并且在这段时间内这些旧数据仍然可以被读取。这使得写入操作和读取操作不必互斥，并使读取历史数据成为可能。
+
+存在超过一定时间并且不再使用的版本会被清理，否则它们将始终占用硬盘空间，并对性能产生负面影响。我们使用一个垃圾回收 (GC) 机制来清理这些旧数据。
+
+## 工作方式
+
+TiDB 会周期性地进行 GC 。每个 TiDB Server 启动后都会在后台运行一个 gc_worker ，每个集群中会有其中一个 gc_worker 被选为 leader ， leader 负责维护 GC 的状态并向所有的 TiKV region leader 发送 GC 命令。
+
+## 配置与监测方法
+
+GC 相关的配置和运行状态记录在 `mysql.tidb` 这张系统表中，可以通过 SQL 语句进行检测和配置：
+
+```sql
+mysql> select VARIABLE_NAME, VARIABLE_VALUE from mysql.tidb;
++-----------------------+------------------------------------------------------------------------------------------------+
+| VARIABLE_NAME         | VARIABLE_VALUE                                                                                 |
++-----------------------+------------------------------------------------------------------------------------------------+
+| bootstrapped          | True                                                                                           |
+| tidb_server_version   | 18                                                                                             |
+| tikv_gc_leader_uuid   | 58accebfa7c0004                                                                                |
+| tikv_gc_leader_desc   | host:ip-172-16-30-5, pid:95472, start at 2018-04-11 13:43:30.73076656 +0800 CST m=+0.068873865 |
+| tikv_gc_leader_lease  | 20180418-11:02:30 +0800 CST                                                                    |
+| tikv_gc_run_interval  | 10m0s                                                                                          |
+| tikv_gc_life_time     | 10m0s                                                                                          |
+| tikv_gc_last_run_time | 20180418-10:59:30 +0800 CST                                                                    |
+| tikv_gc_safe_point    | 20180418-10:58:30 +0800 CST                                                                    |
+| tikv_gc_concurrency   | 1                                                                                              |
++-----------------------+------------------------------------------------------------------------------------------------+
+10 rows in set (0.02 sec)
+```
+
+其中， `tikv_gc_run_interval` ， `tikv_gc_life_time` ， `tikv_gc_concurrency` 这三条记录可以手动配置。其余带有 `tikv_gc` 前缀的记录为当前运行状态的记录， TiDB 会自动更新这些记录，请不要手动修改它们。
+
+`tikv_gc_run_interval` 是 GC 运行时间间隔。 `tikv_gc_life_time` 是历史版本的保留时间，每次进行 GC 时，会清理超过该时间的历史数据。这两项配置不应低于 10 分钟，默认值均为 10 分钟。可以直接用 SQL 修改其数值来进行配置。比如，如果我们想保留一天以内的历史数据，就可以执行：
+
+```sql
+update mysql.tidb set VARIABLE_VALUE = '24h' where VARIABLE_NAME = 'tikv_gc_life_time';
+```
+
+时长字符串的形式是数字后接时间单位的序列，如 `24h`、`2h30m`、`2.5h`。可以使用的时间单位包括 "h"、"m"、"s"。
+
+需要注意的是，在数据更新频繁的场景下如果将 `tikv_gc_life_time` 设置得比较大（如数天甚至数月），可能会有一些潜在的问题：
+
+1. 随着版本的不断增多，数据占用的磁盘空间会随之增加。
+2. 大量的历史版本会在一定程度上导致查询变慢，主要影响范围查询（如 `select count(*) from t`）。
+3. 如果在运行中突然将 `tikv_gc_life_time` 配置调小，可能会导致短时间内大量历史数据被删除，造成 I/O 负担。
```

Comment:
Please use the bulleted list, because the 3 items above have no necessary logic in order. Usually, we use the numbered list for steps.